kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/assertions/nondeterministic/eventually.kt

```diff
@@ -242,7 +248,14 @@ private class EventuallyControl(val config: EventuallyConfiguration) {
       lastDelayPeriod = (timeInMillis() - delayMark).milliseconds
    }
 
-   fun hasAttemptsRemaining() = timeInMillis() < end && iterations < config.retries
+   fun hasAttemptsRemaining() : Boolean {
+      // If the duration is infinite, it is considered as default value, which means it will never stop
+      val isInfinite = config.duration == INFINITE
```

Comment:
Isn't this case already covered by `isWithinDuration`?

Reply:
well 

> Isn't this case already covered by `isWithinDuration`?

`isWithinDuration` only checks time depletion
`isInfinte` checks if `config.duration` if is set to default which is `INFINITE`

Reply:
oh maybe since `Duration.INFINITE` is almost same as `INFINITE` I think I can rollback this expression